terraform/modules/rds_migration/ecs.tf

```diff
@@ -0,0 +1,128 @@
+#tfsec:ignore:aws-ecs-enable-container-insight:TODO CLDC-2542 enable container insights if necessary for logging/monitoring
+resource "aws_ecs_cluster" "this" {
+  #checkov:skip=CKV_AWS_65:TODO CLDC-2542 enable container insights if necessary for logging/monitoring
+  name = "${var.prefix}-db-migration"
+}
+
+resource "aws_ecs_task_definition" "ad_hoc_db_migration" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-ad-hoc-db-migration"
+  cpu                      = var.db_migration_task_cpu
+  execution_role_arn       = var.ecs_task_execution_role_arn
+  memory                   = var.db_migration_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = var.ecs_task_role_arn
+
+  container_definitions = jsonencode([
+    {
+      name      = "db-migration"
+      essential = true
+      image     = var.ecr_repository_url
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.this.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      secrets = [
+        { Name = "DATABASE_URL", valueFrom = var.database_connection_string_arn },
+        { Name = "CF_PASSWORD", valueFrom = aws_secretsmanager_secret.cloudfoundry_password.arn },
+        { Name = "CF_SERVICE", valueFrom = aws_secretsmanager_secret.cloudfoundry_service.arn },
+        { Name = "CF_SPACE", valueFrom = aws_secretsmanager_secret.cloudfoundry_space.arn },

```

Comment:
I felt like these were potentially also sensitive (I was obfuscating all the information needed to interact with the PaaS Production DB) but indeed not as much as username and password. If not a secret, would you be ok with hardcoding these into terraform or were you thinking something else?

Reply:
Possibly hardcoding in terraform, possibly as vars (although that's always a bit annoying to deal with), and passed into the container as env variables rather than secrets. As I say it's not really a problem to do it this way, just adds a manual step of setting them (and checking if they're correct if we need to).

Reply:
I've set them up as vars and updated the documentation to say these should be checked and updated if necessary 👍 (for staging and prod these should be correct already, and for dev we don't expect to use it to for this but I have given a placeholding value for the cf service anyway)